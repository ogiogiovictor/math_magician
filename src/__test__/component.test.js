import React from 'react';
import TestRenderer from 'react-test-renderer';
import Calculator from '../components/calculator/calculator.component';
import Quote from '../pages/quote/quote';

describe('testing react components using jest snapshot', () => {

it('Testing Calculator component', () => {
    const testRenderer = TestRenderer.create(<Calculator />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });

  it('Testing Button component', () => {
    const testRenderer = TestRenderer.create(<Quote />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });

});
