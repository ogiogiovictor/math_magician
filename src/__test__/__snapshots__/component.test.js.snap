// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing react components using jest snapshot Testing Button component 1`] = `
<div>
  There's like ten minutes when it's like, 'Okay, wait, who is this guy again?' And then, you know, I just put on the calculator watch and the glasses, and just be all, you know, inappropriate. And then it just works out fine.
</div>
`;

exports[`testing react components using jest snapshot Testing Calculator component 1`] = `
Array [
  <div
    className="letsCalculate"
  >
    Lets do some calculations
  </div>,
  <div
    className="cover-area"
  >
    <div
      className="output"
    >
      <div
        className="current"
      />
    </div>
    <div
      class="calculator-grid"
    >
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn btn-normal double"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn-normal"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
